recipes/aws-c-compression/all/conanfile.py

```diff
@@ -40,10 +40,10 @@ def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        if Version(self.version) <= "0.2.15":
-            self.requires("aws-c-common/0.8.2", transitive_headers=True, transitive_libs=True)
-        else:
-            self.requires("aws-c-common/0.9.6", transitive_headers=True, transitive_libs=True)
+        if self.version == "0.2.18":
```

Comment:
Can we move this to the `conandata.yml` so that changing this in the future does not change the revision of unrelated recipes?

Reply:
Given the number of aws recipes and the different versions of each one, I believe it is more straightforward to keep the conditional dependencies directly in the recipe, even if changes in the future produce a new recipe revision. However, we would need to reach an agreement here cc/ @czoido @perseoGI